src/wallet/reconstruct_seed_flow.c

```diff
@@ -0,0 +1,293 @@
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "reconstruct_seed_flow.h"
+
+#include "card_flow_reconstruct_wallet.h"
+#include "constant_texts.h"
+#include "sha2.h"
+#include "shamir_wrapper.h"
+#include "ui_screens.h"
+#include "ui_state_machine.h"
+#include "wallet_list.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+extern char *ALPHABET;
+extern char *ALPHA_NUMERIC;
+extern char *NUMBERS;
+extern char *PASSPHRASE;
+extern Wallet_shamir_data wallet_shamir_data;
+extern Wallet_credential_data wallet_credential_data;
+extern Wallet wallet;
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+typedef enum {
+  PASSPHRASE_INPUT = 1,
+  PASSPHRASE_CONFIRM,
+  PIN_INPUT,
+  TAP_CARD_FLOW,
+  RECONSTRUCT_SEED,
+  COMPLETED,
+  COMPLETED_WITH_ERRORS,
+  TIMED_OUT,
+  EARLY_EXIT,
+  EXIT,
+} reconstruct_state_e;
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+/**
+ * @brief This function handles different states of the reconstruct wallet seed
+ * flow
+ *
+ * @param state The current state of the flow which needs to be executed
+ * @param seed_out Pointer to buffer which needs to be populated with the wallet
+ * seed
+ * @return reconstruct_state_e The next state of the flow
+ */
+reconstruct_state_e reconstruct_seed_handler(reconstruct_state_e state,
+                                             uint8_t *seed_out);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+reconstruct_state_e reconstruct_seed_handler(reconstruct_state_e state,
+                                             uint8_t *seed_out) {
+  reconstruct_state_e next_state = EXIT;
+  switch (state) {
+    case PASSPHRASE_INPUT: {
+      if (!WALLET_IS_PASSPHRASE_SET(wallet.wallet_info)) {
```

Comment:
do reset/clear `wallet_credential_data.passphrase` before this check, for sanity.